cmake/Platform/Core/Targets/ArduinoBootloaderBurnTargetCreator.cmake

```diff
@@ -28,93 +28,55 @@ function(create_arduino_bootloader_burn_target TARGET_NAME BOARD_ID PROGRAMMER P
     endif ()
 
     # look at bootloader.file
-    set(BOOTLOADER_FOUND True)
-    if (NOT ${BOARD_ID}.bootloader.file)
-        set(BOOTLOADER_FOUND False)
-        # Bootloader is probably defined in the 'menu' settings of the Arduino 1.6 SDK
-        if (${BOARD_ID}.build.mcu)
-            GET_MCU(${${BOARD_ID}.build.mcu} BOARD_MCU)
-            if (NOT ${BOARD_ID}.menu.cpu.${BOARD_MCU}.bootloader.file)
-                message("Missing ${BOARD_ID}.bootloader.file, not creating bootloader burn target ${BOOTLOADER_TARGET}.")
-                return()
-            endif ()
-            set(BOOTLOADER_FOUND True)
-            set(${BOARD_ID}.bootloader.file ${${BOARD_ID}.menu.cpu.${BOARD_MCU}.bootloader.file)
-        endif ()
-    endif ()
-
-    if (NOT ${BOOTLOADER_FOUND})
+    _get_board_property_if_exists(${BOARD_ID} bootloader.file BOOTLOADER_FILE)
```

Comment:
Again - Are you completely sure that your retrieval function covers all the code above?

Reply:
Same here, in order to get correct bootloader file you need to know board cpu.

Reply:
OK, LGTM

Reply:
so lets merge then ? :)

Reply:
Yes, let's merge!